[DROOLS-6490] fix drools docs to avoid mentioning the KnowledgeHelper… (#3764)
kiegroup/kie-docs: doc-content/drools-docs/src/main/asciidoc/LanguageReference/drl-rules-THEN-methods-variables-ref.adoc

[id='drl-rules-THEN-methods-variables-ref_{context}']

> = Other rule action methods from `drools` and `kcontext` variables

Reviewer: Is `kcontext` an alias for `KnowledgeHelper`?

Author: `kcontext` originally had a slightly different meaning than `drools`, but since Drools 6 they're just aliases. The first still works for backward compatibility reasons, but there are no other reasons to keep it and imo we can get rid of at least in docs.

In addition to the standard rule action methods, the {DECISION_ENGINE} supports methods in conjunction with the predefined `drools` variable that you can also use in rule actions.

> You can use the `drools` variable to call methods from the `org.kie.api.runtime.rule.c` class in your {PRODUCT} distribution, which is also the class that the standard rule action methods are based on. For all `drools` rule action options,

Reviewer: Typo?
```suggestion
You can use the `drools` variable to call methods from the `org.kie.api.runtime.rule.RuleContext ` class in your {PRODUCT} distribution, which is also the class that the standard rule action methods are based on. For all `drools` rule action options,
```

Author: yes, fixed.

ifdef::DM,PAM[]
download the *{PRODUCT} {PRODUCT_VERSION_LONG} Source Distribution* ZIP file from the https://access.redhat.com/jbossnetwork/restricted/listSoftware.html[Red Hat Customer Portal] and navigate to `~/{PRODUCT_FILE}-sources/src/kie-api-parent-$VERSION/kie-api/src/main/java/org/kie/api/runtime/rule/RuleContext.java`.
endif::[]
ifdef::DROOLS,JBPM,OP[]
see the {PRODUCT} https://github.com/kiegroup/droolsjbpm-knowledge/blob/{COMMUNITY_VERSION_BRANCH}/kie-api/src/main/java/org/kie/api/runtime/rule/RuleContext.java[RuleContext.java] page in GitHub.
endif::[]

> The `drools` variable has a few methods providing information about the currently firing rule and the set of facts that activated it:

Reviewer 2: ```suggestion
The `drools` variable contains methods that provide information about the firing rule and the set of facts that activated the firing rule:
```

see the {PRODUCT} https://github.com/kiegroup/drools/blob/{COMMUNITY_VERSION_BRANCH}/drools-core/src/main/java/org/drools/core/spi/KnowledgeHelper.java[KnowledgeHelper.java] page in GitHub.
see the {PRODUCT} https://github.com/kiegroup/droolsjbpm-knowledge/blob/{COMMUNITY_VERSION_BRANCH}/kie-api/src/main/java/org/kie/api/runtime/rule/RuleContext.java[RuleContext.java] page in GitHub.
endif::[]

The following examples are common methods that you can use with the `drools` variable:
The `drools` variable has a few methods providing information about the currently firing rule and the set of facts that activated it:

* `drools.halt()`: Terminates rule execution if a user or application has previously called `fireUntilHalt()`. When a user or application calls `fireUntilHalt()`, the {DECISION_ENGINE} starts in _active mode_ and evaluates rules continually until the user or application explicitly calls `halt()`. Otherwise, by default, the {DECISION_ENGINE} runs in _passive mode_ and evaluates rules only when a user or an application explicitly calls `fireAllRules()`.
* `drools.getWorkingMemory()`: Returns the `WorkingMemory` object.

> * `drools.setFocus( "<agenda_group>" )`: Sets the focus to a specified agenda group to which the rule belongs.

Reviewer: As far as I can see there is no explicit mentioned alternative for these two methods. Is it on purpose? 
Is there a "blessed" way to access working memory object and to set the focus of an agenda group?

Author: All those methods are available through the KieRuntime like the example that Ieft
`drools.getKieRuntime().getAgenda().getAgendaGroup( "CleanUp" ).setFocus();`

> * `drools.getMatch()`: Returns the `Match` that activated the currently firing rule. It contains information that are useful for logging and debugging purposes, for instance `drools.getMatch().getObjects()` returns the list of objects that made the rule to fire in the proper tuple order.

Reviewer 2: ```suggestion
* `drools.getMatch()`: Returns the `Match` that activated the currently firing rule. It contains information that is useful for logging and debugging purposes, for instance `drools.getMatch().getObjects()` returns the list of objects, enabling rule to fire in the proper tuple order.
```

> From the `drools` variable you can also obtain a reference to the `KieRuntime` providing useful methods to interact with the running session like:

Reviewer 2: ```suggestion
From the `drools` variable, you can also obtain a reference to the `KieRuntime` providing useful methods to interact with the running session, for example:
```

> * `drools.getKieRuntime().halt()`: Terminates rule execution if a user or application has previously called `fireUntilHalt()`. When a user or application calls `fireUntilHalt()`, the {DECISION_ENGINE} starts in _active mode_ and evaluates rules continually until the user or application explicitly calls `halt()`. Otherwise, by default, the {DECISION_ENGINE} runs in _passive mode_ and evaluates rules only when a user or an application explicitly calls `fireAllRules()`.

Reviewer 2: ```suggestion
* `drools.getKieRuntime().halt()`: Terminates rule execution if a user or application previously called `fireUntilHalt()`. When a user or application calls `fireUntilHalt()` method, the {DECISION_ENGINE} starts in `active` mode and evaluates rules until the user or application explicitly calls `halt()` method. Otherwise, by default, the {DECISION_ENGINE} runs in `passive` mode and evaluates rules only when a user or an application explicitly calls `fireAllRules()` method.
```
